lib/binary_to_decimal.rb

```diff
@@ -4,6 +4,16 @@
 # The least significant bit is at index 7.
 # Calculate  and return the decimal value for this binary number using
 # the algorithm you devised in class.
+# def binary_to_decimal(binary_array)
+
 def binary_to_decimal(binary_array)
-  raise NotImplementedError
+  i = 7
```

Comment:
Minor suggestion: instead of hard-coding the value of `i` to `7`, consider initializing `i` to `binary_array.length - 1`. That way, your code would work for any length of the array.